src/storage/src/hummock/state_store.rs

```diff
@@ -369,6 +388,25 @@ impl StateStore for HummockStorage {
         }
     }
 
+    fn prefix_scan<R, B>(
+        &self,
+        prefix_hint: Option<Vec<u8>>,
```

Comment:
I think a prefix should be required in perfix_scan interface because it is called "prefix" scan. That being said, I have the following suggestions:
1. Use Vec<u8> directly without Option
2. Name it prefix instead of prefix_hint

Reply:
thanks, fix it at new commit

Reply:
> Name it prefix instead of prefix_hint

`prefix` sounds confusing to me, like this:

> or `prefix = t233`, `range = 4..=9`

Reply:
> > Name it prefix instead of prefix_hint
> 
> `prefix` sounds confusing to me, like this:
> 
> > or `prefix = t233`, `range = 4..=9`
> 
> But actually we are:
> 
> > `prefix = t233`, `range = t2334..=t2339`

I also think that  `prefix = t233`, `range = 4..=9` is more intuitive. If it is hard to support that, let's do an assertion to make sure the `range` has valid start_bound and end_bound with the specified prefix to avoid misusage.

Reply:
Discussed with @Li0k offline. The motivation of having `prefix_hint: Option<Vec<u8>>` is to unify the iterator interfaces so that we don't need to have one interface for prefix iterators and one interface for other iterators. In this case, we agree to avoid introducing a method with "prefix_" in its name and just provide prefix_hint to the existing iterator interfaces:
```
    /// Opens and returns an iterator for given `prefix_hit` and `full_key_range`
    /// Internally, `prefix_hit` will be used to construct `filter_key` Option for `bloom_filter`
    /// filtering and use `full_key_range` for iter The returned iterator will iterate data
    /// based on a snapshot corresponding to the given `epoch`.
    fn iter<R, B>(
        &self,
        prefix_hint: Option<Vec<u8>>,
        key_range: R,
        read_options: ReadOptions,
    ) -> Self::PrefixIterFuture<'_, R, B>
    where
        R: RangeBounds<B> + Send,
        B: AsRef<[u8]> + Send;
  
  fn prefix_scan<R, B>(
        &self,
        prefix_hint: Option<Vec<u8>>,
        key_range: R,
        limit: Option<usize>,
        read_options: ReadOptions,
    ) -> Self::PrefixScanFuture<'_, R, B>
    where
        R: RangeBounds<B> + Send + 'static,
        B: AsRef<[u8]> + Send + 'static;
```

When prefix_hint is Some, the implementation will assert that the given prefix_hint is indeed the prefix of the provided key_range.